PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,515 @@
+
+   # returns enum if node is right or left child
+   # returns None if node is the root
+   def getChildDirection(self, node):
+      if self.isRoot(node):
+            direction = None
+      elif self.isRightChild(node):
+         direction = Child.RIGHT
+      else:
+         direction = Child.LEFT
+      return direction
+
+   # returns the node's parent
+   def getParent(self, node):
+      # is it the root?
+      if self.isRoot(node): return None
+
+      parentIndex = self.getParentIndex(node)
+      return self.nodes[parentIndex]
+
+   def getParentIndex(self, node):
+      # get node's index
+      index = self.getIndex(node)
+      return (index - 1) // 2
+
+   # returns the level of the node
+   def getLevel(self, node):
+      if self.isRoot(node): return 0
+      else: return self.getLevel(self.getParent(node)) + 1
+
+   def getHeight(self, node):
+      if not node: return -1
+
+      rightChild = self.getRightChild(node)
+      leftChild = self.getLeftChild(node)
+
+      return max(self.getHeight(rightChild), self.getHeight(leftChild)) + 1
+
+   # returns the node's left line
+   def getLeftLine(self, node):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("left line"))
+
+      return (lines & nodeFamily).pop()
+
+    # returns the node's left line
+   def getRightLine(self, node):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("right line"))
+
+      return (lines & nodeFamily).pop()
+      
+   # returns a tuple of the left and right child of node
+   def getChildren(self, node):
+      return self.getLeftChild(node), self.getRightChild(node)
+
+   # returns a list of all the nodes that descend from node
+   def getAllDescendants(self, node):
+      if not node: return []
+
+      children = self.getChildren(node)
+      returnList = []
+      if children[0]: returnList.append(children[0])
+      if children[1]: returnList.append(children[1])
+
+      return returnList + self.getAllDescendants(children[0]) + self.getAllDescendants(children[1])
+      
+
+   # ----------- DRAWING METHODS -------------------
+   
+   # monkey patching to allow for circles to be drawn easier
+   def _create_circle(self, x, y, r, **kwargs):
+        return self.create_oval(x-r, y-r, x+r, y+r, **kwargs)
+   Canvas.create_circle = _create_circle
+
+   # Create an arrow to point at a node
+   def createArrow(self, node):
+      arrow = self.canvas.create_line(node.coords[0], node.coords[1] - self.CIRCLE_SIZE - self.ARROW_HEIGHT,
+                                   node.coords[0], node.coords[1] - self.CIRCLE_SIZE, arrow="last", fill='red')
+      return (arrow, )
+
+   # move the arrow to point above the node
+   def moveArrow(self, arrow, node, numSteps= 5):
```

Comment:
This method should have a `sleepTime` parameter like the other moveItems methods.

Reply:
Fixed with commit 3439298